receiver/elasticsearchreceiver/scraper_test.go

```diff
@@ -0,0 +1,346 @@
+				mockClient.On("ClusterHealth", mock.Anything).Return(nil, err404)
+
+				sc := newElasticSearchScraper(zap.NewNop(), createDefaultConfig().(*Config))
+				err := sc.start(context.Background(), componenttest.NewNopHost())
+				require.NoError(t, err)
+
+				sc.client = &mockClient
+
+				_, err = sc.scrape(context.Background())
+				require.True(t, scrapererror.IsPartialScrapeError(err))
+				require.Equal(t, err.Error(), err404.Error())
+
+			},
+		},
+		{
+			desc: "Both node stats and cluster health fails",
+			run: func(t *testing.T) {
+				t.Parallel()
+
+				err404 := errors.New("expected status 200 but got 404")
+				err500 := errors.New("expected status 200 but got 500")
+
+				mockClient := mocks.MockElasticsearchClient{}
+				mockClient.On("NodeStats", mock.Anything, []string{"_all"}).Return(nil, err500)
+				mockClient.On("ClusterHealth", mock.Anything).Return(nil, err404)
+
+				sc := newElasticSearchScraper(zap.NewNop(), createDefaultConfig().(*Config))
+				err := sc.start(context.Background(), componenttest.NewNopHost())
+				require.NoError(t, err)
+
+				sc.client = &mockClient
+
+				m, err := sc.scrape(context.Background())
+				require.Contains(t, err.Error(), err404.Error())
+				require.Contains(t, err.Error(), err500.Error())
+
+				require.Equal(t, m.DataPointCount(), 0)
+			},
+		},
+		{
+			desc: "Cluster health status is invalid",
+			run: func(t *testing.T) {
+				t.Parallel()
+
+				ch := clusterHealth(t)
+				ch.Status = "pink"
+
+				mockClient := mocks.MockElasticsearchClient{}
+				mockClient.On("NodeStats", mock.Anything, []string{"_all"}).Return(nodeStats(t), nil)
+				mockClient.On("ClusterHealth", mock.Anything).Return(ch, nil)
+
+				sc := newElasticSearchScraper(zap.NewNop(), createDefaultConfig().(*Config))
+				err := sc.start(context.Background(), componenttest.NewNopHost())
+				require.NoError(t, err)
+
+				sc.client = &mockClient
+
+				_, err = sc.scrape(context.Background())
+				require.True(t, scrapererror.IsPartialScrapeError(err))
+				require.Contains(t, err.Error(), err.Error())
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.desc, testCase.run)
+	}
+}
+
+func clusterHealth(t *testing.T) *model.ClusterHealth {
+	healthJSON, err := ioutil.ReadFile("./testdata/sample_payloads/health.json")
+	require.NoError(t, err)
+
+	clusterHealth := model.ClusterHealth{}
+	require.NoError(t, json.Unmarshal(healthJSON, &clusterHealth))
+
+	return &clusterHealth
+}
+
+func nodeStats(t *testing.T) *model.NodeStats {
+	nodeJSON, err := ioutil.ReadFile("./testdata/sample_payloads/nodes_linux.json")
+	require.NoError(t, err)
+
+	nodeStats := model.NodeStats{}
+	require.NoError(t, json.Unmarshal(nodeJSON, &nodeStats))
+	return &nodeStats
+}
+
+// TestWriteGolden writes golden metrics to testdata; It should only be used to generate golden metrics when changes to metrics are made.
+func TestWriteGolden(t *testing.T) {
```

Comment:
I don't think it's appropriate to keep this functionality in here. If updates are required to the real tests, it should just require us to temporarily comment out `golden.ReadExpected()` add `golden.WriteExpected(actual)`.

I know this process is not ideal, but I don't think it's "safe" to have this code here. If someone ever removes the skip, then any tests depending on golden files are potentially invalided.

Reply:
My reasoning for having it in the first place was really for a form of "documentation"; If I were new to the project and trying to make changes to this receiver, I'm not sure how easy I'd be able to figure out how to generate the new goldens.

That being said, point taken, it does add a gun to shoot yourself in the foot; I've removed it.

Reply:
I also see we have an issue for this: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6891